### ISSUE TITLE
Macros (ala Carbon or Laravel Collections)

### ISSUE THREAD

**Issue by ChrGriffin** [moneyphp/money#600]
An excellent and helpful new feature would be user-definable macros, similar in concept to [Carbon macros](https://carbon.nesbot.com/docs/#api-macro) or [Laravel Collection macros](https://laravel.com/docs/5.8/collections#extending-collections). I'm considering working on this myself and opening a PR, but wanted to throw the issue out there in case something is in the works already.

**Comment by sstok** [comment]
Other than "nice to have", is a there special need that cannot be solved otherwise? do you have specific example where this would usable (in context, not the technique itself).

**Comment by ChrGriffin** [comment]
Do I have an example that absolutely cannot be solved otherwise? No, definitely not. I can always write service classes and pass Money objects into methods to do what I want. For example, I needed to convert from one currency to another, so I wrote a class to access the fixer.io API and convert one Money object to a new one. Ultimately, the implementation ends up looking like:

```
$usd = Money::USD(200);
$cad = new FixerApi()
    ->from('USD')
    ->to('CAD')
    ->on('2012-01-01 00:00:00')
    ->convert();
```

It's a perfectly manageable implementation. But it would definitely be nice to be able to attach a 'convert' to my Money objects that I could simplify that implementation down further to:

```
$cad = Money::USD(200)->convert('CAD', '2012-01-01 00:00:00');
```

So definitely 'nice to have'. But I struggle to come up with a a scenario where I couldn't solve any such issue in a different way.
